internal/report.go

```diff
@@ -17,5 +31,66 @@ type Report struct {
 	// Network error.
 	Error error `json:"error,omitempty"`
 	// Extra information. Depends on the protocol.
-	Extra string `json:"extra,omitempty"`
+	Extra  string `json:"extra,omitempty"`
+	Format Format
+}
+
+func (r *Report) PrintFormatted(f Format) {
```

Comment:
I think this one (and the ones called here)  should return an `string`, printing it only in the main function.